File: generated-code/typescript/src/model/vec2-int.ts

import { StreamWrapper } from "../stream-wrapper";

export class Vec2Int {
    x: number
    y: number

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }

    static async readFrom(stream: StreamWrapper): Promise<Vec2Int> {
        let x;
        x = await stream.readInt();
        let y;
        y = await stream.readInt();
        return new Vec2Int(x, y)
    }

    async writeTo(stream: StreamWrapper) {
        let x = this.x;
        await stream.writeInt(x);
        let y = this.y;
        await stream.writeInt(y);
    }
}

File: generated-code/codecraft/swift/Sources/TransGenTest/Vec2Int.swift
public struct Vec2Int {
    let x: Int32
    let y: Int32

    static func readFrom<S: InputStream>(_ stream: S) -> Vec2Int {
        var x: Int32
        x = stream.readInt32()
        var y: Int32
        y = stream.readInt32()
        return Vec2Int(x: x, y: y)
    }

    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(x)
        stream.writeInt32(y)
    }
}